testing/python/metafunc.py

```diff
@@ -1252,6 +1252,26 @@ def test_it2(foo):
         reprec = testdir.inline_run("--collect-only")
         assert not reprec.getcalls("pytest_internalerror")
 
+    def test_parametrize_with_dict(self, testdir):
+        p1 = testdir.makepyfile(
+            """
+            import pytest
+
+            @pytest.mark.parametrize('arg1,arg2,expected_id', {
+                "myid_12": [1, 2, "[myid_12]"],
+                "ignored_id": pytest.param(3, 4, "[myid_34]", id="myid_34"),
```

Comment:
We should always warn and use the dict imposed id

Reply:
Is a warning really useful/necessary?

Reply:
This will help in situations with mixed aliasing

The warning cam be skipped if the id is equal or none

We should provide a utility method to drop the id explicitly if necessary

Reply:
> This will help in situations with mixed aliasing

Can you elaborate, please?

I think the typical use case is not to use `pytest.param` here in the first place anyway.

> The warning cam be skipped if the id is equal or none

Sure, but still I think people might actually be more annoyed by the warning then.

Which I think leads to you thinking about:

> We should provide a utility method to drop the id explicitly if necessary

￼How/where would that be used then?  Isn't `_replace` good enough?

Reply:
aliasing -> when reusing predeclared param instances that already have a id, potentially a different one

the warning is absolutely necessary to have an api you cant misuse/misunderstand

a internal method generated by a implementation detail is never enough

Reply:
> aliasing -> when reusing predeclared param instances that already have a id, potentially a different one

Do you have a real-world use case / scenario for this?

> the warning is absolutely necessary to have an api you cant misuse/misunderstand

Well, ok - requires to unroll the loop again though (which is ok).
What should be used here `warnings` or `pytest.warn`?

>>> We should provide a utility method to drop the id explicitly if necessary

>> ￼How/where would that be used then? Isn't `_replace` good enough?
 
> a internal method generated by a implementation detail is never enough

￼You mean `_replace` is not enough?  (it is documented for named tuples, but you mean that us using NamedTuple is the detail?!)

However, how should the method work?  Where is the id dropped?  (currently it gets replaced (i.e. dropped) always (from the existing paramset))
Do you mean there should be a way to _keep_ it? (i.e. not use the one from the dict)